src/apps/answers/service.py

```diff
@@ -1922,9 +1924,26 @@ async def is_flow_finished(self, submit_id: uuid.UUID, answer_id: uuid.UUID) ->
         return self._is_activity_last_in_flow(applet_full, activity_id, flow_id)
 
     async def create_report(
-        self, submit_id: uuid.UUID, answer_id: uuid.UUID | None = None
+        self, submit_id: uuid.UUID, answer_id: uuid.UUID | None = None, activity_id: uuid.UUID | None = None
     ) -> ReportServerResponse | None:
         answers = await AnswersCRUD(self.answers_session).get_by_submit_id(submit_id, answer_id)
+        act_crud = ActivitiesCRUD(self.session)
+        subscale_table_data = []
+        subscale_name = ""
+        if activity_id:
+            activity_data = await act_crud.get_by_id(activity_id)
+            scoring_type = activity_data.scores_and_reports.get("scoring_type", "raw_scores")
+            if scoring_type == "lookup_scores":
+                subscale_name = activity_data.scores_and_reports.get("subscale_name", False)
+                if not subscale_name:
+                    raise NoSubscaleLinked()
+                subscales = activity_data.subscale_setting["subscales"]
+                for subscale in subscales:
+                    if subscale["name"] == subscale_name:
+                        subscale_table_data = subscale.get("subscale_table_data", False)
+                        if not subscale_table_data:
+                            raise SubscaleDoesNotExist()
```

Comment:
These errors should be thrown during applet creation/update when the data is being saved. The data that we read in the `create_report` function should've already been validated.

The function `validate_score_and_sections` is the validator that performs these checks. You can add another condition to this loop to check whether each score has a scoring type, and the `score` scoring type is accompanied by a valid subscale name

https://github.com/ChildMindInstitute/mindlogger-backend-refactor/blob/3b4117c062206cf34444bc90d1d66a80824a5027/src/apps/activities/domain/custom_validation.py#L79-L123

Reply:
Added additional checks to the validation function.